docs/src/modules/components/MarkdownDocs.js

```diff
@@ -65,49 +88,16 @@ const styles = (theme) => ({
   },
 });
 
-function flattenPages(pages, current = []) {
-  return pages.reduce((items, item) => {
-    if (item.children && item.children.length > 1) {
-      items = flattenPages(item.children, items);
-    } else {
-      items.push(item.children && item.children.length === 1 ? item.children[0] : item);
-    }
-    return items;
-  }, current);
-}
-
-// To replace with .findIndex() once we stop IE 11 support.
-function findIndex(array, comp) {
-  for (let i = 0; i < array.length; i += 1) {
-    if (comp(array[i])) {
-      return i;
-    }
-  }
-
-  return -1;
-}
-
 function MarkdownDocs(props) {
-  const {
-    classes,
-    disableAd = false,
-    disableToc = false,
-    markdown: markdownProp,
-    req,
-    reqPrefix,
-    reqSource,
-  } = props;
+  const { classes, disableAd = false, disableToc = false, demos, docs, requireDemo } = props;
 
   const t = useSelector((state) => state.options.t);
 
-  const markdownDocs = useMarkdownDocs({
-    markdown: markdownProp,
-    req,
-    reqPrefix,
-    reqSource,
-  });
-
-  const headers = getHeaders(markdownDocs.markdown);
+  const userLanguage = useSelector((state) => state.options.userLanguage);
+  const { description, location, rendered, title, toc } = docs[userLanguage] || docs.en;
+  if (description === undefined) {
+    throw new Error('Missing description in the page');
```

Comment:
This was previously in useMarkdownDocs which is now `prepareMarkdown`. `prepareMarkdown` can be used for markdown without a description e.g. `backers.md`